src/main/scala/com/linagora/openpaas/gatling/core/AvatarsSteps.scala

```diff
@@ -22,13 +22,22 @@ import io.gatling.core.Predef._
 import io.gatling.http.Predef._
 import io.gatling.http.request.builder.HttpRequestBuilder
 
+import scala.util.Random
+
 object AvatarsSteps {
-  def search(queryKey: String): HttpRequestBuilder = {
-      http("load avatar for user")
-        .get(s"/api/avatars")
-        .queryParam("objectType", "user")
-        .queryParam("email", s"$${$queryKey}")
-      .check(status.in(200, 304)
+  def search(queryKey: String, withRandomDisplayName: Boolean = false): HttpRequestBuilder = {
+    val query = http("load avatar for user" + (if(withRandomDisplayName) " with random display name" else ""))
+      .get(s"/api/avatars")
+      .queryParam("objectType", "user")
+      .queryParam("email", s"$${$queryKey}")
+
+    (if (withRandomDisplayName) {
+      query.queryParam("displayName", _ => Random.alphanumeric.filter(_.isLetter).take(Random.nextInt(10) + 1).toList.mkString
       )
+    }
+    else {
+      query
+    })
+      .check(status.in(200, 304))
```

Comment:
I don't know Scala, but I think it would be simpler (more readable) if you did
```scala
if (withRandomDisplayName) {
  query = query.queryParam("displayName", _ => Random.alphanumeric.filter(_.isLetter).take(Random.nextInt(10) + 1).toList.mkString)
}

query.check(status.in(200, 304))
```

Reply:
> I don't know Scala, but I think it would be simpler (more readable) if you did
> 
> ```scala
> if (withRandomDisplayName) {
>   query = query.queryParam("displayName", _ => Random.alphanumeric.filter(_.isLetter).take(Random.nextInt(10) + 1).toList.mkString)
> }
> 
> query.check(status.in(200, 304))
> ```

I would not work because query is an immutable builder, and then statement in the 'if' would be ignored.

Reply:
So it would not work even using `var` instead of `val`?

Reply:
ah sorry i didn't not see the assignation. it would work with a var. But i stand by that using mutable variable would not be very idiomatic in scala.